x/autopilot/module_ibc.go

```diff
@@ -123,51 +125,83 @@ func (im IBCModule) OnRecvPacket(
 		return channeltypes.NewErrorAcknowledgement(err)
 	}
 
-	// to be utilized from ibc-go v5.1.0
-	if data.Memo == "stakeibc/LiquidStake" {
-		strideAccAddress, err := sdk.AccAddressFromBech32(data.Receiver)
-		if err != nil {
-			return channeltypes.NewErrorAcknowledgement(err)
-		}
-
-		ack := im.app.OnRecvPacket(ctx, packet, relayer)
-		if ack.Success() {
-			return im.keeper.TryLiquidStaking(ctx, packet, data, &types.ParsedReceiver{
-				ShouldLiquidStake: true,
-				StrideAccAddress:  strideAccAddress,
-			}, ack)
-		}
-		return ack
+	// ibc-go v5 has a Memo field that can store forwarding info
+	// For older version of ibc-go, the data must be stored in the receiver field
+	var metadata string
+	if data.Memo != "" { // ibc-go v5+
+		metadata = data.Memo
+	} else { // before ibc-go v5
+		metadata = data.Receiver
```

Comment:
The common case will be where `data.Receiver` is just an address - thoughts on adding an explicit check for that case and short-circuiting the rest of the logic, if it's just an address? A bit redundant, but it protects standard IBC transfers from any of the autopilot logic interfering

Reply:
yeah good call. Something like:
```go
_, err := sdk.AccAddressFromBech32(data.Receiver)
if err == nil && data.Memo == "" {
   return im.app.OnRecPacket(...)
}
```
?

Reply:
done